### PR TITLE
Add BinaryQuickSave Option

### PR DESCRIPTION
**Pull request by przemko07**
Binary saving helps to preserve older save games, while allowing for a frequent saving. If you dedicate 10 slots for saving, it will save 1024 times in a binary theme. The lower save slots will be saved more frequent, then higher save slots.

### CODE REVIEW

File: zUtilities/QuickSave.cpp [hunk 1]
```diff
@@ -59,10 +59,88 @@ namespace GOTHIC_ENGINE {
       return;
     }
 
-    iLastSaveSlot++;
-    iLastSaveNumber++;
-    if ( iLastSaveSlot > Options::MaxSaveSlot || iLastSaveSlot < Options::MinSaveSlot )
-      iLastSaveSlot = Options::MinSaveSlot;
+    if ( Options::UseQuickSave ) {
```
Comment (Franisz) [inline zUtilities/QuickSave.cpp:62]: Should check for `Options::UseBinarySave`

File: zUtilities/QuickSave.cpp [hunk 2]
```diff
@@ -59,10 +59,88 @@ namespace GOTHIC_ENGINE {
...
+
+        // We are saving for the first time probably
+        if ( iLastSaveNumber == Invalid) {
+            iLastSaveSlot = Options::MinSaveSlot;
+            iLastSaveNumber = 1; // I could use ZERO as a first SaveName, it doesnt matter. number ONE is just more pleasant to the eye
+        }
+        // we need to figure out, on which slot to put next save.
+        else {
+            iLastSaveNumber++;
+            int const nextName = maxName + 1;
```
Comment (Franisz) [inline zUtilities/QuickSave.cpp:108]: `maxName` is undefined

File: zUtilities/QuickSave.cpp (hunk 1 above)
Comment (przemko07) [reply zUtilities/QuickSave.cpp]: changed `Options::UseBinarySave` -> `Options::UseQuickSave`

**Comment by przemko07** [comment]
Fixed bitwise operations, totally forgot how to write right shift register `>>` and the bit and operations `&` have higer priority than greater operator `>` :D

**Pull request merged**